src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -119,7 +120,7 @@ def append_user_agent(self, user_agent: str):
         Args:
             user_agent (str): The user agent information to append.
         """
-        if "user_agent" in self:
+        if self.USER_AGENT_KEY in self:
```

Comment:
I think `OperationContext.USER_AGENT_KEY` is better

Reply:
Updated